### PR TITLE
solution

### PR DESCRIPTION
**Pull request by serentinos**
[DEMO LINK](https://serentinos.github.io/react_dynamic-list-of-goods/)

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -1,27 +1,48 @@
...
+  const [goods, setGoods] = useState<Good[]>([]);
 
-    <button type="button" data-cy="all-button">
-      Load all goods
-    </button>
+  const onClickHandle = async (
+    f: () => Promise<Good[]>,
+  ) => {
+    setGoods(await f());
+  };
```
Comment (33Orange) [inline src/App.tsx:15]: Create 3 separate handlers for buttons

```suggestion
  const handleClickAllGoods = (event: ...) => {
    event.preventDefault();
    setGoods(getAll());
  };
```

File: src/App.tsx [hunk 2]
```diff
@@ -1,27 +1,48 @@
...
+    <div className="App">
+      <h1>Dynamic list of Goods</h1>
 
-    <button type="button" data-cy="red-button">
-      Load red goods
-    </button>
+      <button
+        type="button"
+        data-cy="all-button"
+        onClick={() => onClickHandle(getAll)}
```
Comment (33Orange) [inline src/App.tsx:24]:
```suggestion
        onClick={handleClickAllGoods}
```

File: src/api/goods.ts [hunk 3]
```diff
@@ -3,17 +3,27 @@ import { Good } from '../types/Good';
...
+  } catch (error) {
+    throw new Error('Cant Load data');
+  }
 }
 
-export const get5First = () => {
-  return getAll()
-    .then(goods => goods); // sort and get the first 5
+export const get5First = async () => {
+  const response = await getAll();
```
Comment (dffUqp) [inline src/api/goods.ts:18]: What will we do if `getAll` return an error?

**Review by dffUqp — APPROVED** [review]
GJ✔️

one thing to improve the code is to disable the button when data is loading to avoid extra requests to API

**Pull request closed without merging**